### PR TITLE
Game controller handlers

### PR DESCRIPTION
**Pull request by MettyS**

### CODE REVIEW

File: server/controller/central_checkers_controller.go [hunk 1]
```diff
@@ -2,95 +2,108 @@ package controller
...
+// GameList map list of all game instances, pairing a GameID with a game handler
+var GameList map[string]gameplay.Game
 
-// GameStartRequest incoming request to join and start a game
-type GameStartRequest struct {
-	GameID string
+// d.GameStartRequest expected fields:
+// GameID string - optional
+func validateGameStartRequest(req d.GameStartRequest) (bool, string) {
+	return true, ""
```
Comment (jer-irl) [inline server/controller/central_checkers_controller.go:17]: Can you add a TODO comment here?

File: server/controller/central_checkers_controller.go [hunk 2]
```diff
@@ -2,95 +2,108 @@ package controller
...
+func HandleStartGame(req d.GameStartRequest) (d.GameStartResponse, error) {
+	gameID := req.GameID
+	playerID := "RandomKeyChangeThis"
+
+	var playerRole d.GameRole
+	var err error
+	if gameInstance, exists := GameList[gameID]; exists {
+		playerRole, err = gameInstance.AddParticipant(playerID)
+	} else {
+		gameID = "RandomGameIDOverwriteChangeThis"
```
Comment (jer-irl) [inline server/controller/central_checkers_controller.go]: TODO comment, please!

File: server/controller/central_checkers_controller.go [hunk 3]
```diff
@@ -2,95 +2,108 @@ package controller
...
+func HandleBoardUpdateSubscription(req d.BoardSubscriptionRequest) (d.BoardUpdate, error) {
+	return d.BoardUpdate{}, nil
 }
 
 // HandleStartGame TODO
-func HandleStartGame(req GameStartRequest) (GameStartResponse, error) {
-	return GameStartResponse{}, nil
+func HandleStartGame(req d.GameStartRequest) (d.GameStartResponse, error) {
+	gameID := req.GameID
+	playerID := "RandomKeyChangeThis"
```
Comment (jer-irl) [inline server/controller/central_checkers_controller.go]: TODO comment here, please!

File: server/services/gamecontrol.go [hunk 4]
```diff
@@ -12,19 +13,20 @@ type GameControlServer struct {
 	pb.UnimplementedGameControlServiceServer
 }
 
-func convertGameStartRequestInward(req *pb.GameStartRequest) controller.GameStartRequest {
-	domGameStartRequest := controller.GameStartRequest{}
+func convertGameStartRequestInward(req *pb.GameStartRequest) d.GameStartRequest {
+	domGameStartRequest := d.GameStartRequest{}
```
Comment (jer-irl) [inline server/services/gamecontrol.go]: I personally prefer a name that is harder to confuse with "DOM."  Like `domain` as a prefix instead.

**Review by jer-irl — CHANGES_REQUESTED** [review]

File: server/services/gamecontrol.go (hunk 4 above)
Comment (MettyS) [reply server/services/gamecontrol.go]: I refactored this prefix on this advice.

**Review by jer-irl — APPROVED** [review]

**Pull request merged**